### ISSUE TITLE
Upgrading the the latest drupal/core dev branch

### ISSUE THREAD

**Issue by joelpittet** [drupal-composer/drupal-composer-project#243]
Struggling with getting this `drupal/core` upgrade. Currently the `8.x.x` branches are `8.3` is ready for release and in alpha and `8.4.x` branch is open and being committed to.

I've managed to get the `8.3.x` dev branch to install with composer, but it won't let me get to `8.4.x`.

I tried a bunch of things but the one I expect to work is:
```composer update drupal/core:8.4-dev --with-dependencies```

The response from composer is less than helpful and googling hasn't yield any results:
```Package "drupal/core:8.4-dev" listed for update is not installed. Ignoring.```

Why is it `Ignoring`?

I've added `-vvv` and there no more information to be had:
```
Gathering patches for root package.
> pre-update-cmd: DrupalProject\composer\ScriptHandler::checkComposerVersion
Reading ./composer.lock
Package "drupal/core:8.4-dev" listed for update is not installed. Ignoring.
Loading composer repositories with package information
Writing /Users/pittet/.composer/cache/repo/https---packages.drupal.org-8/packages.json into cache
Writing /Users/pittet/.composer/cache/repo/https---packagist.org/packages.json into cache
Updating dependencies (including require-dev)
```

**Comment by webflo** [comment]
"composer update" expects a package name. You can't change the version constraint via "composer update". I tried "composer require drupal/core:^8.4" with and without --update-with-dependencies but it failed. symfony/dom-crawler prevents the installation. I have to look at it again tomorrow.

**Comment by joelpittet** [comment]
Thanks @webflo, I think my testing method was a bit flawed. I was using `drush st` after the upgrade and it was reporting ` Drupal version                  :  8.3.0-dev`  So I thought nothing was happening.  And the status report page is also reporting that...  But the code shows the right version if I looked in `core/lib/Drupal.php`...  

`drush ev "echo \Drupal::VERSION;"` Was also giving me the wrong value:(  I restarted the VM, and it started to work... I think OPcode cache was messing with me.

Geez that was painful.
